File: tests/expected/comb_sort.jl



function comb_sort(seq::Array{Int64})::Array{Int64}
    gap::UInt64 = length(seq)
    swap = true
    while gap > 1 || swap
        n::Int32 = Int64(Int64(floor((gap / 1.25))))
        gap = max(1, n)
        swap = false
        for i in (0:(length(seq)-gap)-1)
            if seq[i+1] > seq[(i+gap)+1]
                seq[i+1], seq[(i+gap)+1] = (seq[(i+gap)+1], seq[i+1])
                swap = true
            end
        end
    end
    return seq
end

function main()
    unsorted = [14, 11, 19, 5, 16, 10, 19, 12, 5, 12]
    expected = [5, 5, 10, 11, 12, 12, 14, 16, 19, 19]
    @assert(comb_sort(unsorted) == expected)
    println(join(["OK"], " "))
end

main()

File: tests/expected/comb_sort.kt
import kotlin.math.floor
import kotlin.math.max

fun comb_sort(seq: Array<Int>): Array<Int> {
    var gap: Int = seq.size
    var swap = true
    while (gap > 1 || swap) {
        var n: Int = floor((gap.toDouble() / 1.25)).toInt().toInt()
        gap = max(1.toDouble(), n.toDouble()).toInt()
        swap = false
        for (i in (0..(seq.size.toInt() - gap) - 1)) {
            if (seq[i] > seq[(i.toInt() + gap)]) {
                if (true) {
                    val (__tmp1, __tmp2) = Pair(seq[(i.toInt() + gap)], seq[i])
                    seq[i] = __tmp1
                    seq[(i.toInt() + gap)] = __tmp2
                }
                swap = true
            }
        }
    }
    return seq
}

fun main(argv: Array<String>) {
    var unsorted = arrayOf(14, 11, 19, 5, 16, 10, 19, 12, 5, 12)
    val expected = arrayOf(5, 5, 10, 11, 12, 12, 14, 16, 19, 19)
    assert(comb_sort(unsorted) == expected)
    println("OK")
}
